Neuro/Vision/VisionHandler.cs

```diff
@@ -0,0 +1,258 @@
+                Info($"{playerLocation.Key.name} has spent {roundTime.Minutes} minutes and {roundTime.Seconds} seconds near me this round ({roundPercentage * 100.0f:0.0}% of the round)");
+            }
+        }
+    }
+
+    public void ResetAfterMeeting()
+    {
+        // Keep track of what time the round started
+        roundStartTime = Time.timeSinceLevelLoad;
+
+        // Reset our count of how much time per round we've spent near each other player
+        foreach (KeyValuePair<PlayerControl, LastSeenPlayer> playerLocation in playerLocations)
+        {
+            if (playerLocation.Key == PlayerControl.LocalPlayer || playerLocation.Value.location == "") continue;
+            playerLocation.Value.roundTimeVisible = 0f;
+        }
+
+        deadBodies.Clear();
+    }
+
+    private void UpdateDeadBodiesVision()
+    {
+        DirectionToNearestBody = Vector2.zero;
+        float nearestBodyDistance = Mathf.Infinity;
+
+        foreach (DeadBody deadBody in deadBodies)
+        {
+            float distance = Vector2.Distance(deadBody.transform.position, PlayerControl.LocalPlayer.transform.position);
+            if (distance < nearestBodyDistance)
+            {
+                nearestBodyDistance = distance;
+                DirectionToNearestBody = (deadBody.transform.position - PlayerControl.LocalPlayer.transform.position).normalized;
+            }
+
+            if (!IsVisible(deadBody.TruePosition))
+            {
+                continue;
+            }
+
+            if (distance < 3f)
+            {
+                PlayerControl playerControl = playerControls[deadBody.ParentId];
+                playerLocations[playerControl].location = GetLocationFromPosition(playerControl.transform.position);
+                if (!playerLocations[playerControl].dead)
+                {
+                    playerLocations[playerControl].time = Time.timeSinceLevelLoad;
+                    List<PlayerControl> witnesses = new();
+                    foreach (PlayerControl potentialWitness in playerControls.Values)
+                    {
+                        if (PlayerControl.LocalPlayer == potentialWitness) continue;
+
+                        if (potentialWitness.inVent || potentialWitness.Data.IsDead) continue;
+
+                        if (Vector2.Distance(potentialWitness.transform.position, deadBody.transform.position) < 3f) witnesses.Add(potentialWitness);
+                    }
+
+                    playerLocations[playerControl].witnesses = witnesses.ToArray();
+                }
+
+                playerLocations[playerControl].dead = true;
+
+                Info(playerControl.name + " is dead in " + GetLocationFromPosition(playerControl.transform.position));
+            }
+        }
+    }
+
+    private void UpdateNearbyPlayersVision()
+    {
+        foreach (PlayerControl playerControl in playerControls.Values)
```

Comment:
Worst case scenario, we:
- access `playerControl.transform.position` 3 times
- compute `playerControl.GetTruePosition()` 3 times
- access `PlayerControl.LocalPlayer.transform.position` once per loop
- compute `PlayerControl.LocalPlayer.GetTruePosition()` twice per loop

When instead these could be stored in variables

Reply:
In order to merge this pull request before the next stream, I will not be updating this method for now. It definitely needs to be updated but I won't be doing it here.